sql/sql_tablespace.cc

```diff
@@ -32,7 +32,7 @@ int mysql_alter_tablespace(THD *thd, st_alter_tablespace *ts_info)
     If the user haven't defined an engine, this will fallback to using the
     default storage engine.
   */
-  if (hton == NULL || hton->state != SHOW_OPTION_YES)
+  if (hton == NULL || !ha_storage_engine_is_enabled(hton))
```

Comment:
We can't get `binlog` here as it is `HTON_NOT_USER_SELECTABLE`. So just remove `hton->state` check.